### PR TITLE
DDCNL-3138: Migrated to bas gateway

### PR DESCRIPTION
**Pull request by struong**

### CODE REVIEW

**Review by jpatton638 — APPROVED** [review]

**Review by jpatton638 — APPROVED** [review]

File: app/config/ConfigDecorator.scala [hunk 1]
```diff
@@ -66,12 +60,11 @@ class ConfigDecorator @Inject()(
   lazy val preferencesFrontendService = decorateUrlForLocalDev(s"preferences-frontend").getOrElse("")
   lazy val contactHost = decorateUrlForLocalDev(s"contact-frontend.host").getOrElse("")
   lazy val citizenAuthHost = decorateUrlForLocalDev(s"citizen-auth.host").getOrElse("")
-  lazy val companyAuthHost = decorateUrlForLocalDev(s"company-auth.host").getOrElse("")
-  lazy val companyAuthFrontendHost = decorateUrlForLocalDev(s"company-auth-frontend.host").getOrElse("")
   lazy val taiHost = decorateUrlForLocalDev(s"tai-frontend.host").getOrElse("")
   lazy val formTrackingHost = decorateUrlForLocalDev(s"tracking-frontend.host").getOrElse("")
 
   lazy val identityVerificationHost = decorateUrlForLocalDev(s"identity-verification.host").getOrElse("")
+  lazy val identityVerificationPrefix = decorateUrlForLocalDev(s"identity-verification.prefix").getOrElse("mdtp")
```
Comment (struong) [inline app/config/ConfigDecorator.scala:67]: @vimalsuresh @anujakhare iv uplift should now go to iv-stubs for local and staging, real iv for everything else

File: app/config/ConfigDecorator.scala [hunk 2]
```diff
@@ -95,7 +88,7 @@ class ConfigDecorator @Inject()(
       .getOrElse("/template/mustache")
 
   def transformUrlForSso(url: URL) =
-    s"$companyAuthFrontendHost/ssoout/non-digital?continue=" + URLEncoder.encode(url.toString, "UTF-8")
+    s"$basGatewayFrontendHost/bas-gateway/ssoout/non-digital?continue=" + URLEncoder.encode(url.toString, "UTF-8")
```
Comment (struong) [inline app/config/ConfigDecorator.scala:91]: @vimalsuresh @anujakhare when logged in as a SA user, going to portal should redirect them to bas-gateway signout first

File: app/config/ConfigDecorator.scala [hunk 3]
```diff
@@ -255,8 +248,8 @@ class ConfigDecorator @Inject()(
   def getFeedbackSurveyUrl(origin: Origin): String =
     feedbackSurveyFrontendHost + "/feedback/" + enc(origin.origin)
 
-  def getCompanyAuthFrontendSignOutUrl(continueUrl: String): String =
-    companyAuthHost + s"/gg/sign-out?continue=$continueUrl"
+  def getBasGatewayFrontendSignOutUrl(continueUrl: String): String =
+    basGatewayFrontendHost + s"/bas-gateway/sign-out-without-state?continue=$continueUrl"
```
Comment (struong) [inline app/config/ConfigDecorator.scala:252]: @vimalsuresh @anujakhare sign out should redirect them to this new bas-gateway url with /sign-out-without-state? in. I have checked the continueUrl parameter is correct with team-gateway

**Review by jpatton638 — APPROVED** [review]

File: app/controllers/SaWrongCredentialsController.scala [hunk 4]
```diff
@@ -48,16 +48,16 @@ class SaWrongCredentialsController @Inject()(
   private val authenticate: ActionBuilder[UserRequest, AnyContent] = authJourney.authWithSelfAssessment
 
   def ggSignInUrl: String = {
-    lazy val ggSignIn = s"${configDecorator.companyAuthHost}/${configDecorator.gg_web_context}"
+    lazy val ggSignIn = s"${configDecorator.basGatewayFrontendHost}/bas-gateway/sign-in"
```
Comment (struong) [inline app/controllers/SaWrongCredentialsController.scala:51]: @vimalsuresh @anujakhare for a SA ambiguous user we ask then to sign in again using this page, /personal-account/self-assessment/sign-in-again 

This should send you to the correct sign-in route with the right continue_url set

File: app/controllers/auth/AuthAction.scala [hunk 5]
```diff
@@ -146,20 +145,20 @@ class AuthActionImpl @Inject()(
 
       request.session.get(configDecorator.authProviderKey) match {
         case Some(configDecorator.authProviderVerify) => {
-          lazy val idaSignIn = s"${configDecorator.citizenAuthHost}/${configDecorator.ida_web_context}/login"
+          lazy val idaSignIn = s"${configDecorator.citizenAuthHost}/ida/login"
```
Comment (struong) [inline app/controllers/auth/AuthAction.scala:148]: @vimalsuresh @anujakhare  check that verify signing in/out works fine

File: app/controllers/auth/AuthAction.scala [hunk 6]
```diff
@@ -146,20 +145,20 @@ class AuthActionImpl @Inject()(
...
           Redirect(idaSignIn).withSession(
             "loginOrigin"    -> configDecorator.defaultOrigin.origin,
             "login_redirect" -> postSignInRedirectUrl(request)
           )
         }
         case Some(configDecorator.authProviderGG) => {
-          lazy val ggSignIn = s"${configDecorator.companyAuthHost}/${configDecorator.gg_web_context}"
+          lazy val ggSignIn = s"${configDecorator.basGatewayFrontendHost}/bas-gateway/sign-in"
```
Comment (struong) [inline app/controllers/auth/AuthAction.scala:155]: @vimalsuresh @anujakhare check that gg signing in/out works fine when you have no active session e.g. you navigate to the PTA page directly and not through auth login stub

**Review by jpatton638 — APPROVED** [review]

File: app/config/ConfigDecorator.scala (hunk 2 above)
Comment (vimalsuresh) [reply app/config/ConfigDecorator.scala:95]: Checked

File: app/config/ConfigDecorator.scala (hunk 3 above)
Comment (vimalsuresh) [reply app/config/ConfigDecorator.scala:256]: 👍

**Pull request merged**